examples/tutorials/online_asr_tutorial.py

```diff
@@ -0,0 +1,161 @@
+# Streaming inference works on input data with overlap.
+# Emformer RNN-T expects right context like illustrated like the
+# following.
+#
+# .. image:: https://download.pytorch.org/torchaudio/tutorial-assets/emformer_rnnt_context.png
+#
+# The size of main segment and right context can be retrieved from bundle.
+#
+
+frames_per_chunk = bundle.segment_length * bundle.hop_length
+right_context_size = bundle.right_context_length * bundle.hop_length
+
+print(f"Main segment size: {frames_per_chunk} frames ({frames_per_chunk / bundle.sample_rate} seconds)")
+print(f"Right context size: {right_context_size} frames ({right_context_size / bundle.sample_rate} seconds)")
+
+######################################################################
+# 4. Configure the audio stream
+# -----------------------------
+#
+# Next, we configure the input audio stream using :py:func:`~torchaudio.prototype.io.Streamer`.
+#
+# For the detail of this API, please refer to the
+# `Media Stream API tutorial <../tutorials/streaming_api_tutorial.html>`__.
+#
+
+src = (
+    "https://ia800302.us.archive.org/33/items/"
+    "great_pirate_stories_0905_librivox.org/greatpiratestories_00_various.mp3"
+)
+
+streamer = Streamer(src)
+streamer.add_basic_audio_stream(frames_per_chunk=frames_per_chunk, sample_rate=bundle.sample_rate)
+
+print(streamer.get_src_stream_info(0))
+print(streamer.get_out_stream_info(0))
+
+######################################################################
+# `Streamer` does not provide right context, so we create a wrapper
+# that caches the segment and append right context.
+
+
+class ContextStream:
+    def __init__(self, streamer, right_context):
+        self.streamer = streamer
+        self.right_context = right_context
+
+    def __iter__(self):
+        streamer = self.streamer.stream()
+        (segment,) = next(streamer)
+        for (next_segment,) in streamer:
+            chunk = torch.cat((segment, next_segment[: self.right_context, :]))
+            segment = next_segment
+            yield chunk.T[0]
+
+
+######################################################################
+# The following is the audio we will stream.
+#
+# .. raw:: html
+#
+#    <audio controls>
+#       <source src="https://ia800302.us.archive.org/33/items/great_pirate_stories_0905_librivox.org/greatpiratestories_00_various.mp3" type="audio/mp3">
+#    Your browser does not support the audio element.
+#    </audio>
+#
+
+######################################################################
+# 5. Run stream inference
+# -----------------------
+#
+# Finally, we run the recognition.
+#
+
+context_streamer = ContextStream(streamer, right_context_size)
+state, hypothesis = None, None
+
+# fmt: off
+with torch.no_grad():
+    for i, chunk in enumerate(context_streamer):
+        features, length = feature_extractor(chunk)
+        hypos, state = decoder.infer(features, length, 10, state=state, hypothesis=hypothesis)
```

Comment:
Here the beam size can be increased to improve the performance, is 20 good for streaming? @hwangjeff

Reply:
10 seemed to work fine. for shorter sequences (as is often the case in streaming use cases), making it larger likely yields gains that are marginal. also, since this is just a tutorial, i'm not too concerned about meticulously tuning the beam size